Userland/Libraries/LibGUI/UndoStack.cpp

```diff
@@ -61,6 +61,14 @@ void UndoStack::push(NonnullOwnPtr<Command>&& command)
     if (m_stack_index > 0) {
         for (size_t i = 0; i < m_stack_index; i++)
             m_stack.remove(0);
+
+        if (m_clean_index.has_value()) {
+            if (m_clean_index.value() < m_stack_index)
+                m_clean_index.clear();
+            else
+                m_clean_index = m_clean_index.value() - m_stack_index;
```

Comment:
I don't understand this line, why does the clean index move when pushing something on the undo stack?

Reply:
That's because we don't append, we prepend commands. 🙈

Reply:
Oh my goodness.. that's so weird! Why are we doing this 😳

Reply:
That's why I said my brain is hurting from working on this 🤣 

- Start (stack_index = 0, clean_index = {})
- Push command (stack_index = 0, clean_index = {})
- Push command (stack_index = 0, clean_index = {})
- Push command (stack_index = 0, clean_index = {})
- Mark stack clean (stack_index = 0, clean_index = 0)
- Undo one command (stack_index = 1, clean_index = 0)
- Undo one command (stack_index = 2, clean_index = 0)
- Push command (stack_index = 0, clean_index = {})

Reply:
Ok yeah then this patch makes sense. We should really fix the stack to grow in the other direction, too! But we can do that separately :)